director/redirect.go

```diff
@@ -343,17 +343,17 @@ func RegisterOrigin(ctx *gin.Context) {
 			return
 		}
 		if !ok {
-			log.Warningf("Origin %v advertised to namespace %v without valid registration\n",
-				ad.Name, namespace.Path)
-			ctx.JSON(400, gin.H{"error": "Origin not authorized to advertise to this namespace"})
+			log.Warningf("%s %v advertised to namespace %v without valid registration\n",
+				sType, ad.Name, namespace.Path)
+			ctx.JSON(400, gin.H{"error": sType + " not authorized to advertise to this namespace"})
 			return
 		}
 	}
 
```

Comment:
When we have `sType` set to the cache, we should call something like this:
```
ok, err := VerifyAdvertiseToken(token, "/caches/$SITENAME")
```
When `sType` is a origin, we should do the for-loop as above

That way, we still make sure the cache is registered (separately, we need to do a validation workflow for registered caches!) but don't need to fake a namespace ad as done previously.